### PR TITLE
refactor: Menu migrate to @rc-component/trigger

### PR DESCRIPTION
**Pull request by MadCcc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
https://github.com/ant-design/ant-design/issues/41249

### 💡 Background and solution
image
干净了

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     - Refactor Menu with `@rc-component/trigger`.   - Remove `rc-trigger` and reduce bundle size.       |
| 🇨🇳 Chinese |   - 使用 `@rc-component/trigger` 重构 Menu 组件。  - 移除 `rc-trigger` 依赖，缩小打包体积。       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at 56fdcd8</samp>

This pull request improves the `Menu` component by updating its dependency `rc-menu` and refining its submenu theme demo.

### <samp>🤖 Generated by Copilot at 56fdcd8</samp>

* Update `rc-menu` dependency to `~9.9.0` for bug fixes and improvements ([link](https://github.com/ant-design/ant-design/pull/42554/files?diff=unified&w=0#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L135-R135))
* Adjust import order of React and antd components to follow ESLint rule `import/order` ([link](https://github.com/ant-design/ant-design/pull/42554/files?diff=unified&w=0#diff-60bb3dd669baaf1a71e19a1d34d778add5d4929e8a09a43de0e287a66dec5b66L1-R4))
* Remove `getPopupContainer` prop from `Menu` component as it is no longer needed after `rc-menu` refactoring ([link](https://github.com/ant-design/ant-design/pull/42554/files?diff=unified&w=0#diff-60bb3dd669baaf1a71e19a1d34d778add5d4929e8a09a43de0e287a66dec5b66L66))

### CODE REVIEW

**Comment by yoyo837** [comment]
在PR里一并缩小limit size？

**Comment by li-jia-nan** [comment]
小这么多……

File: package.json [hunk 1]
```diff
@@ -308,11 +308,11 @@
   "size-limit": [
     {
       "path": "./dist/antd.min.js",
-      "limit": "385 KiB"
+      "limit": "380 KiB"
```
Comment (yoyo837) [inline package.json:311]: 不是有10KB ？ 不多压缩点？留给token用用

Comment (MadCcc) [reply package.json:311]: token 占不了多少吧，能有 1kb 都多了。。

**Comment by afc163** [comment]
应该能关掉不少 issues，可以搜一下关联进来。

**Comment by afc163** [comment]
这个 PR 的 changelog 可以好好写写

1. 用什么方式重构了什么
2. 由此带来的体积变化
3. 最终修复了哪些常见问题

**Review by afc163 — APPROVED** [review]

**Pull request merged**